### ISSUE TITLE
0.7.8 did not fix login issue for "china".

### ISSUE THREAD

**Issue by rikroe** [bimmerconnected/bimmer_connected#220]
0.7.8 did not fix login issue for "china".
HA 0.117.1 did not fix.

_Originally posted by @nichwang88 in https://github.com/bimmerconnected/bimmer_connected/issues/208#issuecomment-719906177_

**Comment by rikroe** [comment]
@nichwang88 are you able to clone this repository? 
If so, please try to adjust the following line to ` if self._region == Regions.CHINA:` https://github.com/bimmerconnected/bimmer_connected/blob/35c98df327edf6c0d152aaff1da432c3585d61bb/bimmer_connected/account.py#L92

Then install it by `pip3 install -e .` and do `bimmerconnected status EMAIL PASSWORD china` to see if it helps.

**Comment by nichwang88** [comment]
sorry, i dont know how to adjust it.
i use HASSOS

**Comment by nichwang88** [comment]
sorry, i dont know how to adjust it.
i use HASSOS

**Comment by rikroe** [comment]
I just released a beta version to PyPI, it would be great if you're able to test it. To do so, please:
- Install the custom component (https://github.com/bimmerconnected/ha_custom_component) to your HomeAssistant (e.g. via __HACS__)
- Go to your HA `config` directory and edit `custom_components/bmw_connected_drive/manifest.json` 
- In this file, change the `requirements` to `bimmer_connected==0.7.9.0b3`
- Restart HA

Does this work for you again?

**Comment by nichwang88** [comment]
still did not work.
i test it with the custom component, and edit the manifest.json.
then i found BMW configure in integrations, and i tried to login with my username (mobile phone NO. with 86 or not), but i could not login in.

```
Logger: bimmer_connected.account
Source: /usr/local/lib/python3.8/site-packages/bimmer_connected/account.py:103 
First occurred: October 31, 2020, 11:49:40 PM (6 occurrences) 
Last logged: 12:05:15 AM

Authentication failed. Maybe your password is invalid?
```

```
Logger: bimmer_connected.account
Source: /usr/local/lib/python3.8/site-packages/bimmer_connected/account.py:104 
First occurred: October 31, 2020, 11:49:40 PM (6 occurrences) 
Last logged: 12:05:15 AM

The BMW Connected Drive portal returned an error: UNAUTHORIZED (received status code 401 and expected 200).
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/bimmer_connected/account.py", line 99, in _get_oauth_token
    response = self.send_request(url, data=data, headers=headers, allow_redirects=False,
  File "/usr/local/lib/python3.8/site-packages/bimmer_connected/account.py", line 153, in send_request
    raise IOError(msg)
OSError: The BMW Connected Drive portal returned an error: UNAUTHORIZED (received status code 401 and expected 200).
```

**Comment by rikroe** [comment]
Just found your hint that it worked on 0.116 - apparently I overlooked it earlier.
Please try `bimmer_connected==0.7.9.0b4` with the same procedure as above!

**Comment by nichwang88** [comment]
it works finally!!
Thanks!!!

**Comment by rikroe** [comment]
Great to hear! Will push this to HA Core accordingly.
However you will probably need to use the custom component until HA 0.118 as only then there will be config flows implemented.
Thanks for your patience and help!

**Comment by nichwang88** [comment]
OK， thanks for your help!

**Comment by nichwang88** [comment]
I found your Latest release 0.7.10, and i tried to edit `manifest.json` to `bimmer_connected==0.7.10`.
Same ERROR came out! 

```
Logger: bimmer_connected.account
Source: /usr/local/lib/python3.8/site-packages/bimmer_connected/account.py:102 
First occurred: 上午7:00:20 (9 occurrences) 
Last logged: 上午7:06:57

Authentication failed. Maybe your password is invalid?
```

```
Logger: bimmer_connected.account
Source: /usr/local/lib/python3.8/site-packages/bimmer_connected/account.py:103 
First occurred: 上午7:00:20 (9 occurrences) 
Last logged: 上午7:06:57

The BMW Connected Drive portal returned an error: UNAUTHORIZED (received status code 401 and expected 200).
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/bimmer_connected/account.py", line 98, in _get_oauth_token
    response = self.send_request(url, data=data, headers=headers, allow_redirects=False,
  File "/usr/local/lib/python3.8/site-packages/bimmer_connected/account.py", line 152, in send_request
    raise IOError(msg)
OSError: The BMW Connected Drive portal returned an error: UNAUTHORIZED (received status code 401 and expected 200).

```

**Comment by rikroe** [comment]
Many thanks for checking this out! There seemed to be some weird stuff going on with merging.

**Comment by nichwang88** [comment]
0.7.11 works again. GOOD!

**Comment by rikroe** [comment]
Closed by #228.
